### PR TITLE
"Nuclear" Status Display mode set when nuclear bomb is armed.

### PR DESCRIPTION
**Pull request by pgmzeta**
## About the PR 
Nuclear bombs now send a radio packet to status displays when armed, telling them to go into a new `nuclear` mode.

This mode places a live-updating countdown timer in front of a pulsing red nuclear symbol background (via the use of a proxy object).

If the status display mode is forced to nuclear through packet hacking (or varedits) and the bomb isn't armed, it will display just the pulsing red nuclear symbol regardless of whether it's a nukie round or not. No sniffing for game modes :)

Enterprising syndicates (or cowardly crewmembers) can change the status displays away to something else after it's changed; it only sends the one packet.

## Why's this needed? 
Makes the station more reactive to big, game-changing events.

image

### CODE REVIEW

**Review by Gannets — APPROVED** [review]
Yeah this looks cool.

**Comment by Cherman0** [comment]
Do the TermOS nukes that some stations have (like in the Donut3 armory) also do this? Seems like they should.

**Comment by pgmzeta** [comment]
> Do the TermOS nukes that some stations have (like in the Donut3 armory) also do this? Seems like they should.

The donut2 research outpost nuke does this for *just* the outpost status displays; I lifted the background from that and hue-shifted it from yellow to red. I think there's some potential unintuitive behavior around multiple nukes fighting over control of the screens, though, that I'm not sure how I would address. I don't want to tackle it in this PR, but it could be a separate one.

File: code/obj/machinery/status_display.dm [hunk 1]
```diff
@@ -49,15 +50,25 @@ TYPEINFO(/obj/machinery/status_display)
 
 	var/repeat_update = FALSE	// true if we are going to update again this ptick
 
+	/// Reference to the nuclear bomb in Nuclear Operatives mode
+	var/obj/machinery/nuclearbomb/the_bomb = null
+
+	/// Proxy object for putting map text on top of overlays
+	var/obj/proxy = new
```
Comment (ZeWaka) [inline code/obj/machinery/status_display.dm:57]: disposing

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/status_display.dm [hunk 2]
```diff
@@ -86,6 +97,10 @@ TYPEINFO(/obj/machinery/status_display)
 		if(!src.net_id)
 			src.net_id = generate_net_id(src)
 
+	disposing()
+		..()
+		src.proxy?.dispose()
```
Comment (ZeWaka) [inline code/obj/machinery/status_display.dm:102]: not quite
```suggestion
	disposing()
		qdel(src.proxy)
		src.proxy = null
		. = ..()
		
```

**Review by ZeWaka — APPROVED** [review]

File: code/obj/machinery/status_display.dm (hunk 2 above)
Comment (pgmzeta) [reply code/obj/machinery/status_display.dm]: why `qdel` instead of `.dispose()` for this use case? because we're in `disposing()`?

Comment (ZeWaka) [reply code/obj/machinery/status_display.dm]: pretty sure our convention is to never call dispose directly, @pali6?

**Review by frawhst — APPROVED** [review]

File: code/obj/machinery/status_display.dm (hunk 2 above)
Comment (ZeWaka) [reply code/obj/machinery/status_display.dm]: pali said yes

**Pull request merged**